Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -208,6 +264,74 @@ void DiscusMultipleScatteringCorrection::getXMinMax(const Mantid::API::MatrixWor
   }
 }
 
+void DiscusMultipleScatteringCorrection::prepareStructureFactor() {
+  // avoid repeated conversion of bin edges to points inside loop by converting to point data
+  convertWsToPoints(m_SQWS);
+  // if S(Q,w) has been supplied ensure Q is along the x axis of each spectrum (so same as S(Q))
+  if (m_SQWS->getAxis(1)->unit()->unitID() == "MomentumTransfer") {
+    auto transposeAlgorithm = this->createChildAlgorithm("Transpose");
+    transposeAlgorithm->initialize();
+    transposeAlgorithm->setProperty("InputWorkspace", m_SQWS);
+    transposeAlgorithm->setProperty("OutputWorkspace", "_");
+    transposeAlgorithm->execute();
+    m_SQWS = transposeAlgorithm->getProperty("OutputWorkspace");
+  } else if (m_SQWS->getAxis(1)->isSpectra()) {
+    // for elastic set w=0 on the spectrum axis to align code with inelastic
+    auto newAxis = std::make_unique<NumericAxis>(std::vector<double>{0.});
+    newAxis->setUnit("DeltaE");
+    m_SQWS->replaceAxis(1, std::move(newAxis));
+  }
+  // avoid repeated conversion of bin edges to points inside loop by converting to point data
+  convertWsToPoints(m_SQWS);
+
+  // generate log of the structure factor to support gaussian interpolation
+  m_logSQ = m_SQWS->clone();
+
+  for (size_t i = 0; i < m_logSQ->getNumberHistograms(); i++) {
+    auto &ySQ = m_logSQ->mutableY(i);
+
+    std::transform(ySQ.begin(), ySQ.end(), ySQ.begin(), [](double d) -> double {
+      if (d == 0.)
+        return -20.0;
+      else
+        return std::log(d);
+    });
+  }
+}
+
+void DiscusMultipleScatteringCorrection::convertWsToPoints(MatrixWorkspace_sptr &ws) {
```

Comment:
IS there a reason not to use the algorithm `ConvertToPointData`?

Reply:
When importance sampling is "on" there is some flat interpolation done later on in the main bit of the algorithm. So when converting a histogram S(Q) to points it's better to store the y values at the x value of the left bin edge rather than the x value of the bin centre (which is what ConvertToPointData does). Without this you get an offset in the results where the S(Q) appears to be offset by half a bin - I must have seen this in some calculations with fat bins.

This was a choice I'd made before this PR. I think it's just shown up in the diff here because I decided to prune down the algorithm's exec method because it was getting quite large so this (pre-existing) code is now in its own function